linalg/sparsemat.cpp

```diff
@@ -513,10 +505,10 @@ void SparseMatrix::SortColumnIndices()
       // wait for it to finish before we can free device temporaries.
       MFEM_STREAM_SYNC;
```

Comment:
According to https://docs.nvidia.com/cuda/cusparse/index.html#cusparsexcsrsort, the routine cusparseXcsrsort supports asynchronous execution if the Stream Ordered Memory Allocator is available.